Tidy up new React + API explorer for mobile (fixes #3607)

thibaudcolas opened the pull request:
Most of the work from #3607 to make the explorer work better (or at all) on mobile, and also faster everywhere.

Edit: not WIP anymore.

jjanssen commented on client/src/components/Explorer/ExplorerPanel.js:
```diff
@@ -6,7 +6,7 @@ import { STRINGS, MAX_EXPLORER_PAGES } from '../../config/wagtailConfig';
 
 import Button from '../Button/Button';
 import LoadingSpinner from '../LoadingSpinner/LoadingSpinner';
-import Transition, { PUSH, POP, FADE } from '../Transition/Transition';
```
If we are not using `FADE` at the moment, we can clean this up from `Transition.js` and `Transition.scss` as well for now.

thibaudcolas replied:
👍

jjanssen commented on gulpfile.js/tasks/styles.js:
```diff
@@ -33,7 +33,7 @@ gulp.task('styles:sass', function () {
         }).on('error', sass.logError))
         .pipe(autoprefixer({
-            browsers: ['last 3 versions', 'not ie <= 8'],
+            browsers: ['last 3 versions', 'ie 11'],
```
🎉

jjanssen commented on client/src/components/Explorer/reducers/nodes.js:
```diff
@@ -15,26 +15,30 @@ const defaultPageState = {
+    state = Object.assign({}, prevState);
     state[payload.id] = Object.assign({}, defaultPageState, state[payload.id]);
-    break;
+    return state;
```
I like this, looks more consolidated.

thibaudcolas commented:
@jjanssen I think I'm done. There are only two things left on #3607 that I'd like to take care of before the next release, but they will be in separate PRs.

Looking over all of my changes, it's a bit messy – I cleaned up as much as I could to keep the reducers small and fast, and also refactored some of the styles (on "mobile" screen sizes mainly).

thibaudcolas commented:
Yes! Merged 🙂 . There is one last PR at #3645, also detailing how much testing I've done so far. I'm quite confident it works (haha), but any further help is very much appreciated.
